reconcile/openshift_base.py

```diff
@@ -1066,3 +1066,53 @@ def user_has_cluster_access(
     """Check user has access to cluster."""
     userkeys = determine_user_keys_for_access(cluster.name, cluster.auth)
     return any((getattr(user, userkey) in cluster_users for userkey in userkeys))
+
+
+def get_namespace_type_overrides(namespace: Mapping) -> dict[str, str]:
+    """Returns a dict with the type overrides defined in a namespace
+
+    :param namespace: the namespace
+    :return: dict with the the type overrides
+    """
+    return {
+        o["resource"]: o["override"]
+        for o in namespace.get("managedResourceTypeOverrides") or []
+    }
+
+
+def get_namespace_resource_types(
+    namespace: Mapping, type_overrides: Optional[dict[str, str]] = None
+) -> list[str]:
+    """Returns a list with the namespace ResourceTypes, with the overrides in place
+
+    :param namespace: the namespace
+    :param type_overrides: dict with the namespace type overrides
+    :return: Definitive managed resource TYPES for a namespace
+    """
+    if not type_overrides:
+        type_overrides = get_namespace_type_overrides(namespace)
+
+    return [
+        type_overrides.get(t, t) for t in namespace.get("managedResourceTypes") or []
+    ]
+
+
+def get_namespace_resource_names(
+    namespace: Mapping, type_overrides: Optional[dict[str, str]] = None
+) -> dict[str, list[str]]:
+    """Returns a list with the namespace ResourceTypeNames, with the overrides in place
+
+    :param namespace: the namespace
+    :param type_overrides: dict with the namespace type overrides
+    :return: Definitive managed resource NAMES for a namespace
+    """
+    if not type_overrides:
+        type_overrides = get_namespace_type_overrides(namespace)
+    rnames: dict[str, list[str]] = {}
+
+    for item in namespace.get("managedResourceNames") or []:
+        kind = type_overrides.get(item["resource"], item["resource"])
+        ref = rnames.setdefault(kind, [])
+        ref += item["resourceNames"]
```

Comment:
can `item["resourceNames"]` be None?

Reply:
It does not make sense as these objects are meant to limit specific names but I think it's not set in the schemas. I'll update the schemas to enforce the definition.